tests/hats_tap/test_tap_schema_db_threading.py

```diff
@@ -0,0 +1,138 @@
+"""Tests for thread-safety of TAPSchemaDatabase."""
+
+import sqlite3
+import threading
+
+import pytest
+from hats_tap.tap_schema_db import TAPSchemaDatabase
+
+
+class TestThreadSafety:
+    """Test that TAPSchemaDatabase is thread-safe."""
+
+    @pytest.fixture
+    def temp_db(self, tmp_path):
+        """Create a temporary database for testing."""
+        db_path = tmp_path / "test_tap_schema.db"
+        db = TAPSchemaDatabase(str(db_path))
+        db.initialize_schema()
+        # Insert some test data
+        db.insert_schema("test_schema", "Test schema")
+        db.insert_table("test_schema", "test_table", "table", "Test table")
+        return db
+
+    def test_multiple_threads_can_query(self, temp_db):
+        """Test that multiple threads can query the database concurrently."""
+        results = []
```

Comment:
I think using lists with append should be totally ok here under CPython’s GIL. That said, if we ever see flaky behavior, it might be worth switching this (and `errors` in `test_context_manager_works_with_threads` below) to a queue.Queue, for a bit more portability across interpreters